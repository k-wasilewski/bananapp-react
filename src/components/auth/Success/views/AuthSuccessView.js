import React, {Component} from 'react';
import {Link} from 'react-router-dom';
import AuthHome from '../../Home/AuthHome';
import {setUsername} from '../../../../redux/actions';
import { connect } from 'react-redux';

class AuthSuccessView extends Component {
    render() {
        const $this = this;

        return (
            <div className='App'>
                <header className='App-header'>
                    <h3> Logged-in as { this.props.username }</h3>
                    <AuthHome username={this.props.username}/>
                    <Link to={{
                        pathname: '/auth/personalBananas',
                        state: {
                            username: $this.props.username
                        }
                    }}>
                        <button variant='outlined'>
                            Personal bananas
                        </button>
                    </Link>
                    <button variant='outlined' onClick={this.props.logout}>
                        Logout
                    </button>
                </header>
            </div>
        )
    }
}

function mapStateToProps(state) {
    return {
        username: state.setUsernameReducer.username,
    };
};

const mapDispatchToProps = {
    setUsername
};

export default connect(mapStateToProps, mapDispatchToProps)(AuthSuccessView);