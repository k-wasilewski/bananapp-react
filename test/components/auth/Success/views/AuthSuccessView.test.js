import React from 'react';
import renderer from 'react-test-renderer';
import store from "../../../../../src/redux/store";
import {BrowserRouter} from "react-router-dom";
import AuthSuccessView from "../../../../../src/components/auth/Success/views/AuthSuccessView";
import {Provider} from "react-redux";

describe("AuthSuccessView rendering specification", () => {
    it('AuthSuccessView is rendered', () => {
        const component = renderer.create(
            <Provider store={store}>
                <BrowserRouter>
                    <AuthSuccessView />
                </BrowserRouter>
            </Provider>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});