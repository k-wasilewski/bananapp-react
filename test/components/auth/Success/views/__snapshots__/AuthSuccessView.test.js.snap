// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthSuccessView rendering specification AuthSuccessView is rendered 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h3>
       Logged-in as 
      0
    </h3>
    <div>
      <header>
        <input
          name="avatar"
          onChange={[Function]}
          type="file"
        />
        <button
          onClick={[Function]}
          type="button"
        >
           Upload 
        </button>
        <div />
        <div />
        <div
          className="previewText image-container"
        >
          Select a jpg image (max filesize 1 MB) to add a banana
        </div>
      </header>
    </div>
    <a
      href="/auth/personalBananas"
      onClick={[Function]}
    >
      <button
        variant="outlined"
      >
        Personal bananas
      </button>
    </a>
    <button
      variant="outlined"
    >
      Logout
    </button>
  </header>
</div>
`;
